Description: Add support for yodl 4 to yodl version check
Author: Axel Beckert <abe@debian.org>
Bugs-Debian: https://bugs.debian.org/865492
Forwarded: https://www.zsh.org/cgi-bin/mla/redirect?WORKERNUMBER=41342

--- a/configure.ac
+++ b/configure.ac
@@ -628,6 +628,7 @@
     case `yodl --version` in
       *"version 2."*) YODL_OPTIONS='-k' ;;
       *"version 3."*) YODL_OPTIONS='-k -L' ;;
+      *"version 4."*) YODL_OPTIONS='-k -L' ;;
     esac
 fi
 AC_SUBST(YODL_OPTIONS)
