commit 811027a22a806f753537abee1575da8714d3fa0a
Author: Barton E. Schaefer <schaefer@zsh.org>
Date:   Sat Jun 27 15:53:21 2015 -0700
Bug-Debian: https://bugs.debian.org/790282

    35642: signal queueing in zcontext_save_partial()

diff --git a/Src/context.c b/Src/context.c
index 1b8741f..2dc8d3b 100644
--- a/Src/context.c
+++ b/Src/context.c
@@ -53,6 +53,8 @@ zcontext_save_partial(int parts)
 {
     struct context_stack *cs;
 
+    queue_signals();
+
     cs = (struct context_stack *)malloc(sizeof(struct context_stack));
 
     if (parts & ZCONTEXT_HIST) {
@@ -67,6 +69,8 @@ zcontext_save_partial(int parts)
 
     cs->next = cstack;
     cstack = cs;
+
+    unqueue_signals();
 }
 
 /* save context in full */
