Description: 35908: fix $((...)) completion in expand-or-complete widget
Origin: commit 771381c2ce0587d825528d49244006edf770fd47
Author: Peter Stephenson <p.w.stephenson@ntlworld.com>
Bug-Debian: https://bugs.debian.org/793168

Index: zsh/Src/Zle/zle_tricky.c
===================================================================
--- zsh.orig/Src/Zle/zle_tricky.c	2015-07-26 19:50:33.237182978 +0200
+++ zsh/Src/Zle/zle_tricky.c	2015-07-26 19:50:33.237182978 +0200
@@ -730,11 +730,12 @@
 		    }
 		}
 	    }
-	    if (lst == COMP_EXPAND_COMPLETE)
+	    if (lst == COMP_EXPAND_COMPLETE) {
 		do {
 		    /* Check if there is a parameter expression. */
 		    for (; *q && *q != String; q++);
-		    if (*q == String && q[1] != Inpar && q[1] != Inbrack) {
+		    if (*q == String && q[1] != Inpar && q[1] != Inparmath &&
+			q[1] != Inbrack) {
 			if (*++q == Inbrace) {
 			    if (! skipparens(Inbrace, Outbrace, &q) &&
 				q == s + zlemetacs - wb)
@@ -778,6 +779,7 @@
 		    } else
 			break;
 		} while (q < s + zlemetacs - wb);
+	    }
 	    if (lst == COMP_EXPAND_COMPLETE) {
 		/* If it is still not clear if we should use expansion or   *
 		 * completion and there is a `$' or a backtick in the word, *
