commit ef080283468cbd7b46ceb19c148cd0b9c4d52356
Author: Barton E. Schaefer <schaefer@zsh.org>
Date:   Sat Jun 27 12:50:40 2015 -0700

    35634: handle failure of gettempname() in namedpipe()

diff --git a/Src/exec.c b/Src/exec.c
index 50a11eb..39d1326 100644
--- a/Src/exec.c
+++ b/Src/exec.c
@@ -4133,6 +4133,10 @@ namedpipe(void)
 {
     char *tnam = gettempname(NULL, 1);
 
+    if (!tnam) {
+	zerr("failed to create named pipe: %e", errno);
+	return NULL;
+    }
 # ifdef HAVE_MKFIFO
     if (mkfifo(tnam, 0600) < 0){
 # else
