Description: Add missing apt purge and autoremove subcommands
Author: Felipe Sateler <fsateler@debian.org>
Bug-Debian: https://bugs.debian.org/808317
Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=808317;att=1;filename=_apt.diff;msg=5
Forwarded: http://www.zsh.org/mla/workers/2015/msg03348.html

--- a/Completion/Debian/Command/_apt	2015-12-03 20:44:53.000000000 -0300
+++ b/Completion/Debian/Command/_apt	2015-12-18 12:36:49.326848831 -0300
@@ -407,10 +407,12 @@
     /$'install\0'/ /$'[^\0]#\0'/ ':packages::_deb_packages "$expl_packages[@]" avail' \# \
     /$'[^\0/]#/'/ /$'[^\0/]#\0'/ ':release name::_apt_releases' \) \| \
     /$'remove\0'/ /$'[^\0]#\0'/ ':packages::_deb_packages "$expl_packages[@]" installed' \# \| \
+    /$'purge\0'/ /$'[^\0]#\0'/ ':packages::_deb_packages "$expl_packages[@]" installed' \# \| \
     /$'upgrade\0'/ \| \
+    /$'autoremove\0'/ \| \
     /$'full-upgrade\0'/ \| \
     /$'edit-sources\0'/ \| \
-    /"[]"/	':argument-1::compadd "$expl_action[@]" list search show update install remove upgrade full-upgrade edit-sources'
+    /"[]"/	':argument-1::compadd "$expl_action[@]" list search show update install remove upgrade full-upgrade edit-sources autoremove purge'
 
   _apt-cmd () {
     local expl_action expl_packages
